Add osg-ce-attributes-generator (SOFTWARE-4895) (#135)
opensciencegrid/osg-configure: README.md

OSG CE Attributes Generator
===========================

The CE Attributes Generator is a standalone tool to create resource catalog attributes for a CE, that will be uploaded to the CE collector.
It is a lightweight alternative to OSG-Configure, and it prints the attributes instead of modifying your CE's config files.

It uses `*.ini` files from OSG-Configure as the source of the data.

Installation
------------

The CE Attributes Generator is typically installed via RPMs from the OSG repositories.
See [OSG documentation for how to enable the repositories](https://opensciencegrid.org/docs/common/yum/);
once the repos are enabled, install by running

    yum install osg-ce-attributes-generator

Note: `osg-ce-attributes-generator` is not available in the OSG 3.5 series.

Configuration
-------------

The CE Attributes Generator uses the same config files that OSG-Configure uses.
See the [31-cluster.ini](#31-clusterini--subcluster-and-resource-entry-sections),
[35-pilot.ini](#35-pilotini--pilot) for attributes.

You will also need `resource` and `resource_group` from the `Site Information` section
([40-siteinfo.ini](#40-siteinfoini--site-information-section)).

> You can also use `--resource` and `--resource-group` on the command line to specify these.

Reviewer: ```suggestion
You can also use `--resource` and `--resource-group` on the command line to specify these
instead.
```

> In addition, you should also have at least one batch system section with the attribute `enabled=True`, e.g.

Reviewer: ```suggestion
In addition, you need at least one batch system section with the attribute `enabled=True`, e.g.
```

What happens if they have more than one batch system?

Author: They don't need a batch system section with enabled=True if they pass --batch-systems on the command line, or if they're using BOSCO and specifying the batch system in `20-bosco.ini`.

If they have multiple batch systems, the `OSG_BatchSystems` attribute ends up being a comma-separated list.

Reviewer: > They don't need a batch system section with enabled=True if they pass --batch-systems on the command line, or if they're using BOSCO and specifying the batch system in 20-bosco.ini.

Hrmm, neither the "should also have" or "need" versions of this sentence communicate that logic so we should probably number the different ways to specify the batch system in order of prio
